### ISSUE TITLE
URL redirect partially broken in v0.6.1

### ISSUE THREAD

**Issue by stvdilln** [hvac/hvac#343]
In v0.6.0  v1/__init__.py there was code:
```
  # NOTE(ianunruh): workaround for https://github.com/ianunruh/hvac/issues/51
        while response.is_redirect and self.allow_redirects:
            url = urljoin(self._url, response.headers['Location'])
            response = self.session.request(method, url, headers=headers,
                                            allow_redirects=False, **_kwargs)
```
If a redirect occurred then urlparse:urljoin was used.  In v0.6.1 urljoin was switched to an internally written function, that just joins strings with a slash. 
```
 @staticmethod
    def urljoin(*args):
        """
        Joins given arguments into a url. Trailing and leading slashes are
        stripped for each argument.
        """

        return '/'.join(map(lambda x: str(x).strip('/'), args))

```
We have a vault cluster where when you hit a non-master node, you get a fully qualified URL.  The new code then redirects to something like https://vault.my.org:8200/https://10.100.10.1:8200/.  The urljoin from url parse has more advanced parsing that looks at the string being appended and if the second string starts with https:// is just uses the second string.
The new code would work for a redirect to a page on same site, but fails on redirects to a new site.

This change appears intentional, but breaks us.  I don't know what this commit fixed:
https://github.com/hvac/hvac/commit/3f08b5f8d599fcf65aa2a4018337dedcac7f3429

**Comment by jeffwecan** [comment]
Thanks for the report @stvdilln! The issue meant to be solved is outlined with a bit more detail in https://github.com/hvac/hvac/issues/205. We'll try to ensure we have a fix in for your use case by the time hvac v0.7.1 is released.

**Comment by jeffwecan** [comment]
@stvdilln: In you case, does the location header also include the path? I.e., would a request to  `https://vault.my.org:8200/secret/some-secret` return a location header of `https://10.100.10.1:8200/secret/some-secret`?
